### DISCUSSION TITLE
In-game leaderboard toggle doesn't work for replays

### DISCUSSION THREAD

**Discussion by Magnus-Cosmos** [ppy/osu#24869] [Q&A]
### Type

Game behaviour

### Bug description

When watching replays, pressing `TAB` does not toggle the in-game leaderboard. I noticed that with Autoplay, it is always enabled, while for replays, it is always disabled. Relevant discussion: #24075

Also related, the in-game leaderboard toggle doesn't work independent of the in-game interface toggle (compared to Stable where the interface and leaderboard are separate).

### Screenshots or videos

https://github.com/ppy/osu/assets/39173355/e114932c-7386-4a8c-bb6b-2c8a2c549374

### Version

2023.914.0-lazer

### Logs

N/A

**Comment by peppy** [comment]
Currently leaderboards are always shown in certain contexts, including replays.

**Reply by Magnus-Cosmos** [reply]
In the video above, the leaderboard isn't shown for replays. Not sure if that's what you meant. I tried watching local replays too and the local leaderboard doesn't show either. Currently the only "replay" that shows a leaderboard in solo is Autoplay.

**Comment by Shavixinio** [comment]
Seems like this is caused by this:
https://github.com/ppy/osu/blob/7b2ca2f388121717a68e3e50551aa287f3b46704/osu.Game/Screens/Play/ReplayPlayer.cs#L101

Setting it to false seems to fix the problems but only for autoplay and normal gameplay, but not for watching a real player's play.

It looks like it was intentional, but I don't really understand the reason behind making it intentionally show the leaderboard all the time, can someone explain? I know this is an old thread but this annoys me every time I watch a replay and I wanted to discuss this again.
